tests/tests.go

```diff
@@ -315,6 +306,40 @@ func changeReplicas(managerName string, numOfReplica int32) error {
 	return nil
 }
 
+func deletePodsFromDeployment(deploymentName string) error {
+	deployment, err := testClient.KubeClient.AppsV1().Deployments(managerNamespace).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	labelSelector := labels.Set(deployment.Spec.Selector.MatchLabels).String()
+	By(fmt.Sprintf("Deleting pods from deployment %s with label selector %s", deploymentName, labelSelector))
+	err = testClient.KubeClient.CoreV1().Pods(managerNamespace).DeleteCollection(context.TODO(), metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: labelSelector})
+	if err != nil {
+		return err
+	}
+
+	deploymentConditionAvailability := func(conditionStatus corev1.ConditionStatus, timeout, interval time.Duration) {
+		By(fmt.Sprintf("Waiting for deployment %s Available condition to be %s", deploymentName, conditionStatus))
+		Eventually(func() bool {
+			deployment, err := testClient.KubeClient.AppsV1().Deployments(managerNamespace).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+			if err != nil {
+				return false
+			}
+			for _, condition := range deployment.Status.Conditions {
+				if condition.Type == v1.DeploymentAvailable {
+					return condition.Status == conditionStatus
+				}
+			}
+
+			return true
+		}, 2*time.Minute, time.Second/10).Should(BeTrue(), "Failed waiting readiness at for deployment:\n%v", string(deploymentName))
+	}
+
+	deploymentConditionAvailability(corev1.ConditionFalse, 2*time.Minute, time.Second/10)
```

Comment:
What if the deployment is ready before getting to this line?

Reply:
restarting pods takes enouhg time to reach this, or this is what I have see on multiple runs

I can go back to check replicas but we can face the same problem and enmascarade a false Readiness.

Reply:
How do you feel about waiting for the "old" pods to be cleared by specific-name, before changing the deployment replicas back to 1?

Reply:
Ahh that's a good one :-) will do thanks!

Reply:
Done